spec/order_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'order'
+
+describe MyOrder do 
+  
+  before do
+    allow(subject).to receive(:send_text)
+  end
+
+  let(:bad_item) { 'DEFINITELY NOT AN ITEM ON THE MENU' }
+  let(:good_item) { :'Spring Rolls' }
+  let(:good_item_price) { 2.99 }
+  context 'When initialized' do 
+    it 'has an empty order basket' do 
+      expect(subject.basket).to be_empty
+    end
+  end
+
+  describe '#order' do 
+    context 'item is on the menu' do 
+      it 'adds the item to the basket' do 
+        subject.order(good_item)
+        expect(subject.basket.find { |item| item[good_item] }).to_not be_nil
+      end
+    end
+
+    context 'item is not on the menu' do 
+      it 'raises an error' do 
+        expect { subject.order(bad_item) }.to raise_error "Item is not on the menu"
+      end
+    end
+
+  end
+
+  describe '#total' do 
+    context 'nothing in the basket' do 
+      it 'returns 0' do 
+        expect(subject.total).to eq 0
+      end
+    end
+    context 'items are in the basket' do 
+      it 'returns the total cost of the order' do 
+        subject.order(good_item)
+        expect(subject.total).to eq good_item_price
+      end
+    end
+  end
+
+  describe '#confirm_order' do 
+    context 'basket is empty' do 
+      it 'raises an error' do 
+        expect { subject.confirm_order }.to raise_error 'Basket is empty, cannot confirm order'
+      end
+    end
+    context 'basket is not empty' do
+      it 'confirms order' do 
+        subject.order(good_item)
+        expect { subject.confirm_order }.not_to raise_error 'Basket is empty, cannot confirm order'
```

Comment:
do you also need this? since you already check if an error is raised before?

Reply:
good point! I think what I am trying to achieve here is:

- first test checks that if you confirm order when the basket is empty you get an error. 
- second test checks that if you confirm the order when the basket is not empty you dont also get an error. 

So I wanted to check that an error isnt always raised but I see you point that maybe it is too much.